htdocs/core/lib/company.lib.php

```diff
@@ -53,38 +53,72 @@ function societe_prepare_head(Societe $object)
 	if (empty($conf->global->MAIN_SUPPORT_SHARED_CONTACT_BETWEEN_THIRDPARTIES)) {
 		if (empty($conf->global->MAIN_DISABLE_CONTACTS_TAB) && $user->rights->societe->contact->lire) {
 			//$nbContact = count($object->liste_contact(-1,'internal')) + count($object->liste_contact(-1,'external'));
-			$nbContact = 0; // TODO
-
-			$sql = "SELECT COUNT(p.rowid) as nb";
-			$sql .= " FROM ".MAIN_DB_PREFIX."socpeople as p";
-			$sql .= " WHERE p.fk_soc = ".$object->id;
-			$resql = $db->query($sql);
-			if ($resql) {
-				$obj = $db->fetch_object($resql);
-				if ($obj) $nbContact = $obj->nb;
+			$nbContact = 0;
+			if (!empty($conf->memcached->enabled) && !empty($conf->global->MEMCACHED_SERVER)) {
```

Comment:
You don't have to test is a cache exists or not. There is already ready to use function. All you have to do is use
dol_setcache() to cahce the result of the select to count contact 
and use dol_getcache() to get value in cache.
If there is no cache handled, the   dol_getcache wil return nothing.

Reply:
@eldy I did the update